### PR TITLE
[automower] Adapt login due to Husqvarna API change

### LINKED ISSUE #12980: [Automower] Binding does not work due to change in Husqvarna API Authentication
The Automower Connect API Authentication does not work anymore as they moved to a new authentication method. (AppKey and AppSecret).

Example of the new API: 

`curl -X POST -d "grant_type=client_credentials&client_id=XXX_Application_Key_XXX&client_secret=XXX_Applications_Secret_XXX" https://api.authentication.husqvarnagroup.dev/v1/oauth2/token` 

Apparently many/all(?) users are affected: https://community.openhab.org/t/automower-binding-g2/2854/102

## Expected Behavior
The Autower Addon should use the supported authentication method.

## Current Behavior
The Automower Addon does not use a supported authentication method and the connectivity fails and no bridge can be configured successfully.

## Possible Solution

Add a field for App Secret and change the Authentication method.

Working example: `curl -X POST -d "grant_type=client_credentials&client_id=XXX_Application_Key_XXX&client_secret=XXX_Applications_Secret_XXX" https://api.authentication.husqvarnagroup.dev/v1/oauth2/token` 

## Steps to Reproduce (for Bugs)

1. Create a new Automower Bridge
2. Use Valid credentials + Valid API Key
3. Bridge does not work and error from Husqvarna API is being logged

> 2022-06-20 22:16:28.815 [ERROR] [oauth2client.internal.OAuthConnector] - grant type password to URL 
> https://api.authentication.husqvarnagroup.dev/v1/oauth2/token failed with error code unauthorized_client, description Unauthorized client: `grant_type` is invalid
> 2022-06-20 22:16:28.824 [WARN ] [ternal.bridge.AutomowerBridgeHandler] - Unable to fetch automowers: Rest call failed: statusCode=-1, message=Unable to authenticate
> 2022-06-20 23:16:29.134 [ERROR] [oauth2client.internal.OAuthConnector] - grant type password to URL https://api.authentication.husqvarnagroup.dev/v1/oauth2/token failed with error code unauthorized_client, description Unauthorized client: `grant_type` is invalid

## Your Environment
Using OpenHAB 3.2.0 but also tried the development branch on Github of the openhab-addons (=3.3.0 afaik) with a fresh OpenHAB 3.3.0 installation from the website.

## Notes

I tried to fix it myself but I couldn't get it to work with the OAuth Implementation provided by OpenHAB (the extension uses the built-in OAuth service) as I could not get it to change the grant_type. If there are any pointers how this could be achieved without rolling a new OAuth implementation myself, I'm happy to try to contribute. 

I am unsure if the 'old' username/password combination needs to be removed or if it needs to stay there for compatibility reasons.

### PR DESCRIPTION
**Pull request by crnjan**
The Automower Connect API Authentication does not work anymore as they moved to a new authentication method (AppKey and AppSecret) - adopted. 

Fix #12980

Signed-off-by: Boris Krivonog [boris.krivonog@inova.si](mailto:boris.krivonog@inova.si)

### CODE REVIEW

**Comment by lolodomo** [comment]
This is a breaking change as thing parameters are removed/added. To be mentioned in the release notes through a separate PR.

**Comment by jlaur** [comment]
For reference: Similar to #13050 (same API change).

File: bundles/org.openhab.binding.automower/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -18,13 +18,11 @@
...
-			<parameter name="userName" type="text" required="true">
-				<label>User Name</label>
-				<description>The user name for which the application key has been issued</description>
-			</parameter>
-			<parameter name="password" type="text" required="true">
-				<context>password</context>
-				<description>The password for the given user</description>
+			<parameter name="appSecret" type="text" required="true">
+				<label>Application Secret</label>
+				<description>The Application Secret is required to communication with the Automower Connect Api at
```
Comment (jlaur) [inline bundles/org.openhab.binding.automower/src/main/resources/OH-INF/thing/thing-types.xml:23]:
```suggestion
				<description>The Application Secret is required to communicate with the Automower Connect API at
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thank you for making the binding work again! I have found only a small issue with a text.

File: bundles/org.openhab.binding.automower/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (crnjan) [reply bundles/org.openhab.binding.automower/src/main/resources/OH-INF/thing/thing-types.xml]: Thanks for checking, fixed [existing text](https://github.com/openhab/openhab-addons/pull/13263/files#diff-7c4fc533f8b23716c87311aae0f6095f1c88b7e526f10f11e4325a76abbc2217L14) also ...

Comment (jlaur) [reply bundles/org.openhab.binding.automower/src/main/resources/OH-INF/thing/thing-types.xml]: Thanks, I missed that. Now the I18N properties file must also be regenerated.

Comment (crnjan) [reply bundles/org.openhab.binding.automower/src/main/resources/OH-INF/thing/thing-types.xml]: Done.

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by jlaur** [comment]
@crnjan - as @lolodomo mentioned, it would be much appreciated if you could create a PR towards the openhab-distro repository with a new section and note for 3.4 about the changed configuration parameter:
https://github.com/openhab/openhab-distro/blob/5635e95a953930c2ef01640b392cb8d2a9af4887/distributions/openhab/src/main/resources/bin/update.lst#L77

Although it's debatable whether it's a breaking change (since the binding was already broken by the API change), it's still a change that users need to make in order to get the binding working again.

**Comment by crnjan** [comment]
@jlaur - created https://github.com/openhab/openhab-distro/pull/1413, fyi.

**Pull request merged**